apps/dev-recruiters/src/pages/dashboard.tsx

```diff
@@ -229,26 +254,42 @@ const Dashboard: React.FC = () => {
           <hr />
           <div className="grid grid-cols-1 md:grid-cols-3 gap-8 mt-8">
             {archivedRoles.map((position, index) => (
-              <ArchivedRole
-                key={position.id || index}
-                role={position.title || 'Unknown Role'}
-                department={
-                  teamNames.length > 0
-                    ? `Your Team: ${teamNames.join(', ')}`
-                    : 'Unknown Department'
-                }
-                date={position.postedDate || 'N/A'}
-                onView={() => console.log('View Archived:', position)}
-              />
-            ))}
-            {archivedRoles &&
-              archivedRoles.filter(
-                (result) => result.source === 'opportunities'
-              ).length === 0 && (
-                <div className="col-span-full">
-                  <p>No archived roles available.</p>
+              <div className="flex flex-col" key={String(position.id ?? index)}>
+                <ArchivedRole
+                  key={String(position.id ?? index)}
+                  role={position.attributes.title || 'Unknown Role'}
+                  department={
+                    teamNames.length > 0
+                      ? `Your Team: ${teamNames.join(', ')}`
+                      : 'Unknown Department'
+                  }
+                  date={
+                    typeof position.attributes.published_at === 'string'
```

Comment:
The backend should return a single type. We shouldn't need to figure out the type.